test/e2e/util_gatewayapi_test.go

```diff
@@ -941,3 +944,52 @@ func (m *vapManager) enable() {
 		m.t.Errorf("failed to find vap %q: %v", m.name, err)
 	}
 }
+
+func eventuallyClusterRoleContainsAggregatedPolicies(t *testing.T, destClusterRoleName, srcClusterRoleName string) error {
+	t.Helper()
+
+	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		var destClusterRole rbacv1.ClusterRole
+		if err := kclient.Get(context.Background(), types.NamespacedName{Name: destClusterRoleName}, &destClusterRole); err != nil {
+			if kerrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
```

Comment:
We usually retry on all errors, and in this case, I think it is notable and worth logging if you get a not-found error for any of the standard clusterroles.  
```suggestion
			t.Logf("Failed to get destination ClusterRole %s; retrying...: %v", err)
			return false, nil
```